### PR TITLE
Patch GetSpecificApiVersionOnCondition to respect minimum API version (client.APIVersion)

### PR DESCRIPTION
**Pull request by Didainius**
This is a continuation of #639.
The function `GetSpecificApiVersionOnCondition` did not respect minimum API version. This PR fixes the function so that we don't have errors waiting to happen like it was in #636 case. Also it simplifies API calls that used `GetSpecificApiVersionOnCondition` that are not relevant anymore.

### Note

When `client.APIVersion` is bumped, this function may use newer API version than the constraints defined therefore one must test the code.

### Testing

Tests passed on 10.5.1.1 with only known anomalies:
```
FAIL: org_saml_test.go:112: TestVCD.TestClient_RetrieveRemoteDoc
FAIL: vm_test.go:2047: TestVCD.Test_GetOvfEnvironment
FAIL: nsxt_l2_vpn_tunnel_test.go:16: TestVCD.Test_NsxtL2VpnTunnel
```

### CODE REVIEW

**Review by adambarreiro — APPROVED** [review]
Thanks, LGTM

**Review by dataclouder — APPROVED** [review]

File: govcd/api_vcd_versions.go [hunk 1]
```diff
@@ -231,13 +231,23 @@ func (client *Client) validateAPIVersion() error {
...
-// described in vcdApiVersionCondition
-// f.e. values ">= 32.0", "32.0" returns 32.0 if vCD version is above or 9.7
-func (client *Client) GetSpecificApiVersionOnCondition(vcdApiVersionCondition, wantedApiVersion string) string {
+// GetSpecificApiVersionOnCondition returns default version or wantedApiVersion if it is connected
+// to version described in vcdApiVersionCondition e.g. values ">= 32.0", "32.0" returns 32.0 if vCD
+// version is above or 9.7
+// Note. This function will always respect minimum supported API version which is defined in
+// client.APIVersion. If the wantedApiVersionOrMinimumRequired is less than minimum supported
+// version, this function will return the minimum supported version
+func (client *Client) GetSpecificApiVersionOnCondition(vcdApiVersionCondition, wantedApiVersionOrMinimumRequired string) string {
```
Comment (lvirbalas) [inline govcd/api_vcd_versions.go:240]: Would be useful to have a paragraph explaining the risk involved here (connecting to a newer API version than originally developed against).

Comment (Didainius) [reply govcd/api_vcd_versions.go:242]: Added

**Comment by Didainius** [comment]
> Thank you. Please add a note for reference to PR description too (about the risk involved in using this function without retesting on newer VCD versions).

Added

**Pull request merged**